src/dtemplate.d

```diff
@@ -208,28 +208,36 @@ extern (C++) Expression getValue(Expression e)
  */
 extern (C++) bool match(RootObject o1, RootObject o2)
 {
-    Type t1 = isType(o1);
-    Type t2 = isType(o2);
-    Dsymbol s1 = isDsymbol(o1);
-    Dsymbol s2 = isDsymbol(o2);
-    Expression e1 = s1 ? getValue(s1) : getValue(isExpression(o1));
-    Expression e2 = s2 ? getValue(s2) : getValue(isExpression(o2));
-    Tuple u1 = isTuple(o1);
-    Tuple u2 = isTuple(o2);
+    enum debugPrint = 0;
+
+    auto t1 = isType(o1);
```

Comment:
Interesting. I opened a new PR #5276 to do the small optimization. Then, I'll remove the `match()` changes from this PR.